src/core/ddsi/include/dds/ddsi/q_whc.h

```diff
@@ -45,7 +45,7 @@ struct whc_state {
    an iter on the stack without specifying an implementation. If future changes or
    implementations require more, these can be adjusted.  An implementation should check
    things fit at compile time. */
-#define WHC_SAMPLE_ITER_SIZE (9 * sizeof(void *))
+#define WHC_SAMPLE_ITER_SIZE (7 * sizeof(void *))
```

Comment:
I do remember changing it from 8 to 9, and I think that was it failed to build on some 32-bit platforms with the introduction of the proxy topic iteration — some combination of platform and build flags that just happened not to run on the CI.

Obviously just tracking an entity id is going to require less state than the list iterator, so I'm not surprised it can shrink a bit. But now I am a little nervous about the `7` where it used to be `8`, perhaps you could double-check that the one case that forced it to grow to `9` is not affected by this?

Reply:
Good catch, that was PR #1295, and I've checked it with a local x86 build, that fails with `7`, so I'll change it to `8` (which matches with the change in `bwhc_iter` member from `proxy_topic_list_iter_t` to `ddsi_entityid_t`, one pointer is removed from the type).